### ISSUE TITLE
[health:1.0.2] after releasing app in google play, we found some crashes from health package

### ISSUE THREAD

**Issue by mr-mamali** [cph-cachet/flutter-plugins#151]
Hi There, we have used this package with version 1.0.2 , some few bugs was fixed by changing package source and finally we release our application in play store .

after few days we have found some crash reports and ANR(s) from google play console which device name and android version and stack trace reports from google play have been posted below .

i didn't find the issue to resolve it, if you please check this stack trace report from google play

Thank You alot ..

**Aug 6, 6:23 PM on app version 23
Huawei 荣耀 8X (HWJSN-H), 5888MB RAM, Android 10
Report 2 of 2**

```
java.util.concurrent.ExecutionException: 
  at com.google.android.gms.tasks.Tasks.zzb (Tasks.java:31)
  at com.google.android.gms.tasks.Tasks.a (Tasks.java:31)
  at cachet.plugins.health.HealthPlugin$getData$1.b (HealthPlugin.java:67)
  at cachet.plugins.health.HealthPlugin$getData$1.invoke (HealthPlugin.java)
  at cachet.plugins.health.HealthPlugin$getData$1.invoke (HealthPlugin.java)
  at kotlin.concurrent.ThreadsKt$thread$thread$1.run (ThreadsKt.java:2)
Caused by: com.google.android.gms.common.api.ApiException: 
  at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus (ApiExceptionUtil.java:14)
  at com.google.android.gms.common.internal.zai.zaf (zai.java)
  at com.google.android.gms.common.internal.zaj.onComplete (zaj.java:32)
  at com.google.android.gms.common.api.internal.BasePendingResult.b (BasePendingResult.java:77)
  at com.google.android.gms.common.api.internal.BasePendingResult.a (BasePendingResult.java:41)
  at com.google.android.gms.common.api.internal.BaseImplementation$ApiMethodImpl.setFailedResult (BaseImplementation.java:15)
  at com.google.android.gms.common.api.internal.zae.a (zae.java:2)
  at com.google.android.gms.common.api.internal.GoogleApiManager$zaa.a (GoogleApiManager.java:27)
  at com.google.android.gms.common.api.internal.GoogleApiManager$zaa.a (GoogleApiManager.java:164)
  at com.google.android.gms.common.internal.zag.onConnectionFailed (zag.java:2)
  at com.google.android.gms.common.internal.BaseGmsClient$zzf.zza (BaseGmsClient.java:14)
  at com.google.android.gms.common.internal.BaseGmsClient$zza.a (BaseGmsClient.java:45)
  at com.google.android.gms.common.internal.BaseGmsClient$zzc.zzo (BaseGmsClient.java:51)
  at com.google.android.gms.common.internal.BaseGmsClient$zzb.handleMessage (BaseGmsClient.java:264)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at com.google.android.gms.internal.common.zze.dispatchMessage (zze.java)
  at android.os.Looper.loop (Looper.java:213)
  at android.os.HandlerThread.run (HandlerThread.java:67)
```

**Yesterday, 6:21 PM on app version 23
Huawei Mate 10 (HWALP), 5888MB RAM, Android 10
Report 1 of 2**

```
java.util.concurrent.ExecutionException: 
  at com.google.android.gms.tasks.Tasks.zzb (Tasks.java:31)
  at com.google.android.gms.tasks.Tasks.a (Tasks.java:31)
  at cachet.plugins.health.HealthPlugin$getData$1.b (HealthPlugin.java:67)
  at cachet.plugins.health.HealthPlugin$getData$1.invoke (HealthPlugin.java)
  at cachet.plugins.health.HealthPlugin$getData$1.invoke (HealthPlugin.java)
  at kotlin.concurrent.ThreadsKt$thread$thread$1.run (ThreadsKt.java:2)
Caused by: com.google.android.gms.common.api.ApiException: 
  at com.google.android.gms.common.internal.ApiExceptionUtil.fromStatus (ApiExceptionUtil.java:14)
  at com.google.android.gms.common.internal.zai.zaf (zai.java)
  at com.google.android.gms.common.internal.zaj.onComplete (zaj.java:32)
  at com.google.android.gms.common.api.internal.BasePendingResult.b (BasePendingResult.java:77)
  at com.google.android.gms.common.api.internal.BasePendingResult.a (BasePendingResult.java:41)
  at com.google.android.gms.common.api.internal.BaseImplementation$ApiMethodImpl.setFailedResult (BaseImplementation.java:15)
  at com.google.android.gms.common.api.internal.zae.a (zae.java:2)
  at com.google.android.gms.common.api.internal.GoogleApiManager$zaa.a (GoogleApiManager.java:27)
  at com.google.android.gms.common.api.internal.GoogleApiManager$zaa.a (GoogleApiManager.java:164)
  at com.google.android.gms.common.internal.zag.onConnectionFailed (zag.java:2)
  at com.google.android.gms.common.internal.BaseGmsClient$zzf.zza (BaseGmsClient.java:14)
  at com.google.android.gms.common.internal.BaseGmsClient$zza.a (BaseGmsClient.java:45)
  at com.google.android.gms.common.internal.BaseGmsClient$zzc.zzo (BaseGmsClient.java:51)
  at com.google.android.gms.common.internal.BaseGmsClient$zzb.handleMessage (BaseGmsClient.java:264)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at com.google.android.gms.internal.common.zze.dispatchMessage (zze.java)
  at android.os.Looper.loop (Looper.java:213)
  at android.os.HandlerThread.run (HandlerThread.java:67)

```

**Comment by thomasnilsson** [comment]
Hard to say. Try using the newest version (2.0.1 at the current time) and see if the problem persists. 1.0.2 was released in March and is therefore pretty outdated at this point

**Comment by bardram** [comment]
Fixed (addressed) as part of PR #483 and released as version 3.4.1. 

Thanks to @matsunanaro for the PR.
